src/sql/workbench/services/query/common/resultSerializer.ts

```diff
@@ -170,22 +178,21 @@ export class ResultSerializer {
 		return fileFilters;
 	}
 
-	public getBasicSaveParameters(format: string): SaveResultsRequestParams {
-		let saveResultsParams: SaveResultsRequestParams;
-
-		if (format === SaveFormat.CSV) {
-			saveResultsParams = this.getConfigForCsv();
-		} else if (format === SaveFormat.JSON) {
-			saveResultsParams = this.getConfigForJson();
-		} else if (format === SaveFormat.EXCEL) {
-			saveResultsParams = this.getConfigForExcel();
-		} else if (format === SaveFormat.XML) {
-			saveResultsParams = this.getConfigForXml();
+	public getBasicSaveParameters(format: SaveFormat): SaveResultsRequestParams {
+		switch (format) {
```

Comment:
is this not going to cause linting error? not all path returns a value.

Reply:
I don't get one ... I suppose it's because `format` is an enum, as long as all the enumeration values are accounted for, it's fine? I don't know what should be returned or thrown in the "default" (and basically unreachable) case